feature_engine/datetime/datetime.py

```diff
@@ -250,13 +271,32 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         # Check if input data contains same number of columns as dataframe used to fit.
         _check_input_matches_training_df(X, self.n_features_in_)
 
+        # special case index
+        if self.variables_ is None:
+            if self.missing_values == "raise":
+                self._check_index_contains_na(X.index)
+
+            for feat in self.features_to_extract_:
+                X[FEATURES_SUFFIXES[feat][1:]] = FEATURES_FUNCTIONS[feat](
+                    pd.Series(
+                        pd.to_datetime(
+                            X.index,
+                            dayfirst=self.dayfirst,
+                            yearfirst=self.yearfirst,
+                            utc=self.utc,
+                        ),
+                        index=X.index,
+                    )
+                )
+
+            return X
+
         # check if dataset contains na
         if self.missing_values == "raise":
             _check_contains_na(X, self.variables_)
 
         # reorder variables to match train set
         X = X[self.feature_names_in_]
```

Comment:
This line should also apply for the index features. Is the final of the transform checks, before starting any transform logic. It ensures the df passed to transform has the columns in the same order than the one passed in fit.

Reply:
Right. It probably didn't matter as far as the test go since we're not ever changing the existing columns but the index only when variables="index"